File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def convert_time(time_list): # 문자열"hour:minute"을 분 단위로 변환
    hour, minute = time_list.split(":")
    return int(hour) * 60 + int(minute)

def solution(book_time):
    answer = 0
    
    rooms = [0 for _ in range(24*60 + 10)]
    
    for booked in book_time:
        start = convert_time(booked[0])
        end = convert_time(booked[1]) + 10
        
        rooms[start] += 1
        rooms[end] -= 1
    
    # 누적합 구하기
    for i in range(1, len(rooms)):
        rooms[i] = rooms[i-1] + rooms[i]
    answer = max(rooms)
    return answer

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.java
import java.util.*;

class Solution {
    public int solution(String[][] book_time) {
        int answer = 0;
        
        int MAX = 24 * 60 + 10;
        int[] rooms = new int[MAX+2];
        
        for (String[] booked: book_time) {
            int start = convertTime(booked[0]);
            int end = convertTime(booked[1]) + 10;
            rooms[start] += 1;
            rooms[end] -= 1;
        }
        
        for (int i = 1; i < rooms.length; i++) {
            rooms[i] = rooms[i-1] + rooms[i];
            answer = Math.max(answer, rooms[i]);
        }
        
        return answer;
    }
    
    int convertTime(String stringTime) {
        String[] time = stringTime.split(":");
        int hour = (Integer.parseInt(time[0])) * 60;
        int minute = Integer.parseInt(time[1]);
        return hour + minute;
    }
}